tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc

```diff
@@ -810,20 +813,22 @@ Status AMDGPUTargetModuleLinker(llvm::Module* module, GpuVersion gpu_version,
 // upstream commit), the following mapping will need to change
 std::string MapGCNArchNameTokenToFeatureStr(const std::string& token) {
   if (token == "sramecc+") {
-    return "+sram-ecc";
+    return "+sramecc";
   } else if (token == "sramecc-") {
-    return "-sram-ecc";
+    return "-sramecc";
   } else if (token == "xnack+") {
     return "+xnack";
   } else if (token == "xnack-") {
     return "-xnack";
   }
   return "";
+
 }
 
-std::string GetFeatureStrFromGCNArchName(const std::string& gcn_arch_name) {
+std::pair<std::string, std::string> GetFeatureStrFromGCNArchName(
+    const std::string& gcn_arch_name) {
   std::string feature_str;
-
+  std::string gfx;
```

Comment:
assuming the intent here is to return the "gfxXXX" string in pair.first, then simply initialize `gfx` to `gcn_arch_name` here, and that will make it work for all ROCm versions.

you already have a `if(tokens.size()>0)` condition below that will set `gfx` correctly for when `gcn_arch_name` has feature strings